src/imitation/algorithms/mce_irl.py

```diff
@@ -487,9 +503,11 @@ def train(self, max_iter: int = 1000) -> np.ndarray:
                 predicted_r_np, visitations = self._train_step(torch_obs_mat)
 
                 # these are just for termination conditions & debug logging
-                grad_norm = util.tensor_iter_norm(
-                    p.grad for p in self.reward_net.parameters()
-                ).item()
+                grads = []
```

Comment:
I guess you could also just cast it and keep it was it was. But I don't really feel strongly about it, how it's now seems fine too. Just not obvious why we are doing it, so maybe add a comment saying it's for the type checker.

Reply:
Gradients really can be none in general, but it's true we know they're not here because we called `loss.backward()` in the `self._train_step` call, so a cast does seem fine. I've added a comment for now, but happy to change it to cast, I was pretty on the fence about it.

Reply:
Yeah makes sense. I think the assert is slightly safer, and if there's a comment it should be OK.